### ISSUE TITLE
Cannot read property addEventListener of null

### ISSUE THREAD

**Issue by gorgorgordon** [Kerman46/Cannot-read-property-addEventListener-of-null#1]
The error message says

`Uncaught TypeError: Cannot read property 'addEventListener' of null
    at main.js:9`

Which means that the error occurs at **line 9** of *main.js* of original version.

**Comment by gorgorgordon** [comment]
TL;DR

Rename the id of the second button from siUpIn to signUp.



Video explanation:

https://youtu.be/SM5Wwm7ZMmI



Explaining:

The usual problem occurs with accessing DOM before created, which I have explained in my video https://www.youtube.com/watch?v=HnlzbyLoDRw

As you have added the *defer* keyword in your `<script>` tag at HTML line 8, so the reason is no longer caused by the above.

Otherwise the error would have occurred at line 5 instead, where there is the first addEventListener.


So we look at line 9, 

`signUpButton.addEventListener('click', () => {`

Here this **signUpButton** is *null*, while **signInButton** is not *null*.


So we look into assignment of **signUpButton**, which is at line 2 of **main.js**

`const signUpButton = document.getElementById('signUp');`

Here we are trying to find an element with the id **signUp**.


Now, we look into the HTML, at line 33 of sign.html, the code is:

`<button id="siUpIn">sign up</button>`


Now, we see the mismatch in id to find and the actual id name,

after renaming, the problem is solved.

**Comment by Kerman46** [comment]
Thank you so much for your help, you helped me a lot. I really Appreciate
it.

**Comment by gorgorgordon** [comment]
You are most welcomed ^^

The key is to look into the line according to the line number in error message, and then trace back step by step.

Great start, Kerman. Keep up 💪
